pkg/crc/services/dns/dns_darwin.go

```diff
@@ -60,20 +60,36 @@ func createResolverFile(InstanceIP string, path string) (bool, error) {
 	return true, nil
 }
 
-func restartNetwork(serviceConfig services.ServicePostStartConfig) (bool, error) {
+func restartNetwork() (bool, error) {
 	// https://medium.com/@kumar_pravin/network-restart-on-mac-os-using-shell-script-ab19ba6e6e99
 	netDeviceList, _, err := crcos.RunWithDefaultLocale("networksetup", "-listallnetworkservices")
 	netDeviceList = strings.TrimSpace(netDeviceList)
 	if err != nil {
 		return false, err
 	}
 	for _, netdevice := range strings.Split(netDeviceList, "\n")[1:] {
-		time.Sleep(2)
-		_, stderr, err := crcos.RunWithDefaultLocale("networksetup", "-setnetworkserviceenabled", netdevice, "off")
-		_, stderr, err = crcos.RunWithDefaultLocale("networksetup", "-setnetworkserviceenabled", netdevice, "on")
+		time.Sleep(1 * time.Second)
+		stdout, stderr, err := crcos.RunWithDefaultLocale("networksetup", "-setnetworkserviceenabled", netdevice, "off")
+		logging.DebugF("Disabling the %s Device (stdout: %s), (stderr: %s)", netdevice, stdout, stderr)
+		stdout, stderr, err = crcos.RunWithDefaultLocale("networksetup", "-setnetworkserviceenabled", netdevice, "on")
+		logging.DebugF("Enabling the %s Device (stdout: %s), (stderr: %s)", netdevice, stdout, stderr)
 		if err != nil {
 			return false, fmt.Errorf("%s: %v", stderr, err)
 		}
 	}
+	waitForNetwork()
 	return true, err
 }
+
+// Wait for Network wait till the network is up, since it is required to resolve external dnsquery
+func waitForNetwork() {
+	cmd := "ping -c4 1.1.1.1"
```

Comment:
why we use this????!!!??? this is not guaranteed to work.
We should be checking the users gateway.